drivers/i2s/i2s_nrfx.c

```diff
@@ -27,12 +27,12 @@ struct i2s_nrfx_drv_data {
 	const uint32_t *last_tx_buffer;
 	enum i2s_state state;
 	enum i2s_dir active_dir;
+	bool stop;       /* stop after the current (TX or RX) block */
```

Comment:
Should not it rather be atomic_t? otherwise I doubt it will change anything.

Reply:
I don't see a need for `atomic_t` here. Both `stop` and `discard_rx` are set to `false` only when a transfer is started (i.e. when the interrupt is disabled), then they can be only set to `true` (if the transfer is to be finished for some reason). The problem is that when these members are bit fields, the change made in the interrupt to one of them could potentially overwrite changes done in a thread to some other bit field contained in the same memory unit, as modification of a bit field is a read-modify-write operation on that memory unit. Hence the change.